File: sorts/selectionsort.js
const arr = [64, 25, 12, 22, 11];

function selectionSort (arr) {
    let length = arr.length;

    for(let i = 0; i < length; i++){
        let min = i;

        for(let j = i + 1; j < length; j++) {
            if (arr[j] < arr[min]) {
                min=j;
            }
        }
        if (min != i) {
            let temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
    }
    return arr;
}

console.log(arr);
console.log(selectionSort(arr));

File: sorts/c++/selectionsort.cpp
#include <bits/stdc++.h>
#include <vector>

using namespace std;

int main ()
{
    vector<int> nums{5, 2, -10, 2, 12};
    int length = nums.size();

    for(int i = 0; i < length; i++)
    {
        int min = i;
        for(int j = i + 1; j < length; j++)
        {
            if (nums[j] < nums[min])
            {
                min = j;
            };
        };

        if(min != i)
        {
            int temp = nums[i];
            nums[i] = nums[min];
            nums[min] = temp;
        };
    };

    for(int i = 0; i < length; i++)
    {
        cout << nums[i] << endl;
    };

    return 0;
};
